crates/oxc_isolated_declarations/src/function.rs

```diff
@@ -42,6 +42,54 @@ impl<'a> IsolatedDeclarations<'a> {
         }
     }
 
+    fn copy_binding_pattern_kind_without_assignments(
```

Comment:
I feel that this approach is not very satisfying as any changes to the AST will require updating this code. A simpler approach might be to copy the entire kind and then modify the copy in place using a visitor, but it doesn't seem like this is possible with the visitor/walker abstractions I found in the code.

Reply:
You can do this in a new structure and impl `VisitMut` for it. This way you can modify the AST

Reply:
Updated code to use the visitor approach.